### PR TITLE
rubber refactoring

### PR DESCRIPTION
**Pull request by antoine-de**
remove curl dependency

add some tests

### CODE REVIEW

File: libs/mimir/src/rubber.rs [hunk 1]
```diff
@@ -76,38 +112,33 @@ impl Rubber {
 
     fn create_index(&self, name: &String) -> Result<(), String> {
         debug!("creating index");
-        // Note: for the moment I don't see an easy way to do this with rs_es
+        // Note: fin rs_es it can be done with MappingOperation but for the moment I think
```
Comment (pbougue) [inline libs/mimir/src/rubber.rs]: fin -> in

File: libs/mimir/src/rubber.rs [hunk 2]
```diff
@@ -76,38 +112,33 @@ impl Rubber {
...
-            .and_then(|res| {
-                if res.get_code() == 200 {
-                    Ok(())
-                } else {
-                    Err(format!("cannot create index: {}",
-                                ::std::str::from_utf8(res.get_body()).unwrap()))
-                }
-            })
+        self
+                .put(name, &analysis)
```
Comment (pbougue) [inline libs/mimir/src/rubber.rs]: did you pass fmt? alignment seems weird, but...

File: tests/tests.rs [hunk 3]
```diff
@@ -67,10 +83,21 @@ impl<'a> ElasticSearchWrapper<'a> {
     }
 
     pub fn new(docker_wrapper: &DockerWrapper) -> ElasticSearchWrapper {
-        let es_wrapper = ElasticSearchWrapper { docker_wrapper: docker_wrapper };
+        let mut es_wrapper = ElasticSearchWrapper {
+            docker_wrapper: docker_wrapper,
+            rubber: mimir::rubber::Rubber::new(&docker_wrapper.host())
+         };
```
Comment (pbougue) [inline tests/tests.rs]: fmt

File: libs/mimir/src/rubber.rs [hunk 4]
```diff
@@ -52,62 +55,87 @@ fn get_main_index(doc_type: &str, dataset: &str) -> String {
...
+        let url = self.es_client.full_url(path);
+        let result = try!(self.http_client
+                              .put(&url)
+                              .body(body)
+                              .send());
+        rs_es::do_req(result)
+    }
+    fn post(&self, path: &str, body: &str) -> Result<hyper::client::response::Response, EsError> {
+        // Note: a bit duplicate on rs_es because some ES operations are not implemented
+        info!("doing a post on {} with {}", path, body);
```
Comment (kinnou02) [inline libs/mimir/src/rubber.rs]: debug

**Pull request merged**